slitlessutils/core/modules/extract/single/boxcar.py

```diff
@@ -0,0 +1,175 @@
+    '''
+    Worker function to perform boxcar or "aperture" spectrscopy
+
+    Parameters
+    ----------
+    source : `slitlessutils.core.sources.Source`
+       The source for which to do the extraction
+
+    det : `slitlessutils.core.wfss.data.WFSSDetector()`
+       The data for which to do the extraction
+
+    sci : `np.ndarray`
+       The science image to do the extraction on.
+
+    unc : `np.ndarray`
+       The uncertainty image to do the extraction on.
+
+    dqa : `np.ndarray`
+       The data-quality image to do the extraction on.
+
+    flatfield : `slitlessutils.core.wfss.config.Flatfield`
+       The flat-field object to use.
+
+    order : `slitlessutils.core.wfss.config.Order`
+       The spectral order to use.
+
+    odt : `slitlessutils.core.tables.ODT`
+       The object-dispersion table (ODT).
+
+    width : int or float
+       The extraction width in pixels.  Default is 15 pix
+
+    summation : str
+       The mode of the pixel summation.  Default is 'cartesian'.
+
+    plot : bool
+       Flag to make a diagnostic plot.  Default is False
+
+    Returns
+    -------
+    spectrum : `SpectralTable`
+       The measured spectrum with ancillary data.
+
+    '''
+
+    ny, nx = sci.shape
+
+    fluxscale = Config().fluxscale
+
+    # distill out properties
+    xg = odt.get('x') + 1
+    yg = odt.get('y') + 1
+    vg = odt.get('val')
+    wav = odt.wavelengths()
+
+    msk = np.zeros_like(sci)
+
+    # make an output data structure to fill
+    spec = SpectralTable(odt.segid)
+
+    # half width of the aperture
+    half = (width - 1) / 2
+
+    # get the center
+    x0, y0 = det.ad2xy(*source.adc)
+
+    # in fixed stripes (ie. Cartesian)
+    if summation == 'cartesian':
+        # find unique pixels
+        xi = indices.reverse(xg)
+        # yi = indices.reverse(yg)
+
+        ytrace = np.zeros(len(xi))
+        xtrace = np.zeros(len(xi))
+        for i, (x, idx) in enumerate(xi.items()):
+
+            # collect the data for this slice
+            # xu = xg[idx]
+            yu = yg[idx]
+            vu = vg[idx]
+            wu = wav[idx]
+
+            # compute some things for this slice
+            wave = np.average(wu, weights=vu)
+            yave = np.average(yu, weights=vu)
+            ycent, yfrac = np.divmod(yave, 1)
+            ytrace[i] = yave
+            xtrace[i] = x
+
+            # get the range
+            ymin = max(int(ycent - half), 0)
+            ymax = min(int(ycent + half) + 1, ny - 1)
+
+            # create dummy arrays to vectorize calculations
+            ys = np.arange(ymin, ymax + 1, 1, dtype=int)
+            xs = np.full_like(ys, x, dtype=int)
+
+            # make extraction aperture (always length = 1+width)
+            aper = np.ones_like(ys, dtype=float)
+            aper[0] = 1 - yfrac
+            aper[-1] = yfrac
+            npix = np.sum(aper)
+            if 7550 < wave < 7900:
+                msk[ys, xs] = aper
```

Comment:
```suggestion
            if 7550 < w_avg < 7900:
                msk[ys, xs] = aper
```
Would you explain what this is doing/why this is needed (and maybe add a comment to the code)? If this is a wavelength range, perhaps this should be using Astropy quantities in case the input is in different units than expected here (Angstrom?). Unless there's already code elsewhere that standardizes the units of the input to this function.